gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -947,9 +938,15 @@ private long computeCompletenessWatermark(String table, SortedSet<ZonedDateTime>
         if (timestampDT.isAfter(prevWatermarkDT)
             && TimeIterator.durationBetween(prevWatermarkDT, now, granularity) > 0) {
           long timestampMillis = timestampDT.toInstant().toEpochMilli();
-          if (auditCountVerifier.get().isComplete(table,
-              TimeIterator.dec(timestampDT, granularity, 1).toInstant().toEpochMilli(), timestampMillis)) {
+          ZonedDateTime auditCountCheckLowerBoundDT = TimeIterator.dec(timestampDT, granularity, 1);
+          if (auditCountVerifier.get().isComplete(topicName,
+                  auditCountCheckLowerBoundDT.toInstant().toEpochMilli(), timestampMillis)) {
             completionWatermark = timestampMillis;
+            // Also persist this into State object to share this with other MetadataWriters
+            // we enforce ourselves to always use lower-cased table name here
+            String tableNameLowerCased = tableName.toLowerCase(Locale.ROOT);
```

Comment:
Do we need db name information?

Reply:
this table name already includes catalog and db name, it's in the format of `catalog.db.table`

Reply:
Then change it to be dbTableName to make it more clear?